ffi/tests/ctests/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+#[no_mangle]
+unsafe extern "C" fn rust_assert(check: bool, msg: *const std::os::raw::c_char) {
+    assert!(check, "{}", std::ffi::CStr::from_ptr(msg).to_str().unwrap());
+}
+
+macro_rules! ffi_ctest {
+    (@inner $($name:ident => $link_name:expr,)*) => {
+        $(
+            #[test]
+            pub fn $name() {
+                #[link(name = "wirefilter_ffi")]
```

Comment:
So, apparently, this is also not very cross-platform because generated filename of a Rust library filename can differ between targets.

Instead of relying on it, can you add a regular extern crate above like

```rust
extern crate wirefilter_ffi as _;
```

(`as _` just to mark it as used to avoid warnings)

Reply:
Yep its only possible since we added the `rlib` crate-type. `cdylib` cannot be imported as rust module.

Reply:
Yeah, I think that was the initial problem that Rust tried to warn about (that it can't really link non-Rust binary).

Reply:
With the `extern crate wirefilter_ffi as _;` instead of `#[link]`, I wonder if we are really using the shared library and not the rlib? Using the later might hide some errors that would occur only when linking with the `.so` ?

Reply:
> Using the later might hide some errors that would occur only when linking with the .so?

I would hope not, because at that point it should be just linking differences, not code ones...

Reply:
Done!